### ISSUE TITLE
IOS - problem configuring App Group

### ISSUE THREAD

**Issue by tverilytt** [j3k0/cordova-plugin-openwith#11]
Hi!

Plugin works great in Android, thanks a lot 👍 

I do struggle with IOS though, when in XCode (9.2) configuring the App Group.

I get the exact same message as shown in the screenshot here:

- Add App Group to your App ID (Operation requires bundle identifier)
https://stackoverflow.com/questions/48326175/add-app-group-to-your-app-id-operation-requires-bundle-identifier

I have configured the App Group in IOS Developer pages, adding it to the App ID, and updating the provisioning profile. And doing the same for the "ShareExt" target.

Anyone else run into this situation?!

Cheers
-jo

**Comment by j3k0** [comment]
Hi!

Sorry, no clues here, don't remember having this issue... I recorded the whole procedure for iOS on camera: https://www.youtube.com/watch?v=eaE4m_xO1mg&feature=youtu.be

Great to hear it works on Android though.

**Comment by tverilytt** [comment]
@j3k0 Thanks for that! I have watched the recording, great thing to provide 👍 
Are you on the same XCode version?
Will do some more retries and config checks, seems like XCode does not recognize the relation between the App ID and the App Group, although all looks good in the Web Developer console...

Cheers
-jo

**Comment by j3k0** [comment]
The video is a few month old (October). I'm generally up-to-date with XCode versions, but I don't remember what was the latest at the time.

Good luck. Let me know if you learn anything worth adding to the documentation.

**Comment by tverilytt** [comment]
I finally managed to resolve this, should never underestimate a night session with a couple of beers ;-)

I had counted on that App ID with associated App Groups config, was included in provisioning profile.
But I started to suspect it was not, and therefore tried to, in XCode, to:
1 Modify the "Bundle identifier"
2. Do a Clean and Build
3. Modify the "Bundle identifier" back to original, and correct, name.
4 Do a Clean and Build.

*then* the error message in Capabilities -> App Groups disappeared.

But still, after deploying the app to a device, the share with the app did not appear.

Turned out the "Deployment info" settings for "ShareExt" target was not set for Deployment Target and Devices was set to IPhone. These were modified to match the settings in the app target:
- Deployment Target: 9.0
- Devices: Universal

*then* the share with choice for the app appeared when sharing an image from another app, and all now seems to wrk well, as for Android :-)

Cheers
-jo
